subsys/bluetooth/controller/hci/hci.c

```diff
@@ -35,6 +35,8 @@
 #include "hci_internal.h"
 #include "hci_vendor.h"
 
+#include "ll_sw/ctrl.h"
```

Comment:
Add conditional compilation CONFIG_BT_LL_SW around this include to avoid unnecessary inclusion when building LL_SW_SPLIT

Reply:
I see you point, but this breaks builds and checks start failing, I had to remove this again.

Reply:
hmmm... this only means your changes (to new architecture) depend on old architecture files, this will make it difficult to remove ctrl.h file in the future.